### PR TITLE
Run check & test jobs on Linux, macOS, & Windows with stable, beta, & nightly Rust

### PR DESCRIPTION
**Pull request by autarch**
This would be faster (and less repetitive) if I combined the check & test jobs in one but I didn't want to change too much at first. If you want, I'm happy to make that change.

For some reason this didn't run in Actions when I pushed it to my fork, so I'm not sure it works yet. Maybe it will run when I create this PR.

### CODE REVIEW

**Comment by autarch** [comment]
I'd also note that if you want a really extensive change I can easily add a lot more platforms by using [actions-rust-cross](https://github.com/marketplace/actions/build-rust-projects-with-cross) (which I wrote). This would allow testing on many different Linux CPU architectures.

**Comment by autarch** [comment]
BTW, the reason I started working on this is that I'm seeing an odd failure from this crate when running tests of my own project on Windows. I think the handling of local time zones on that platform is different.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,23 +7,51 @@ on:
...
+    name: check - ${{ matrix.platform.os_name }} with rust ${{ matrix.toolchain }}
+    runs-on: ${{ matrix.platform.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - os_name: Linux
+            os: ubuntu-latest
+          - os_name: macOS
+            os: macOS-latest
```
Comment (waltzofpearls) [inline .github/workflows/ci.yml:19]: `macOS-latest` -> `macos-latest`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -7,23 +7,51 @@ on:
...
+          - os_name: Windows
+            os: windows-latest
+        toolchain:
+          - stable
+          - beta
+          - nightly
```
Comment (waltzofpearls) [inline .github/workflows/ci.yml:25]: Let's have `stable` and `nightly`. I think adding `beta` might be risky as it would increase chance of failures.

**Review by waltzofpearls — COMMENTED** [review]
Hey @autarch, thanks for the PR! It's a great improvement to CI workflows.

I have not tested this library on windows, so this new change will make good coverage across three different operating systems.

I left 2 comments. One for changing all the `macOS-latest` to `macos-latest`, and another for removing `beta` from the list.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (autarch) [reply .github/workflows/ci.yml]: Done.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (autarch) [reply .github/workflows/ci.yml]: Done.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -7,23 +7,50 @@ on:
...
+    name: test - ${{ matrix.platform.os_name }} with rust ${{ matrix.toolchain }}
+    runs-on: ${{ matrix.platform.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - os_name: Linux
+            os: ubuntu-latest
+          - os_name: macOS
+            os: macOS-latest
```
Comment (waltzofpearls) [inline .github/workflows/ci.yml:42]: Could you update this to `macos-latest`?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -7,23 +7,50 @@ on:
...
+          - os_name: Windows
+            os: windows-latest
+        toolchain:
+          - stable
+          - beta
```
Comment (waltzofpearls) [inline .github/workflows/ci.yml:47]: Also remove `beta` from this list.

**Review by waltzofpearls — COMMENTED** [review]
Thanks for the update!

I had another look, and left 2 comments. Same comments as before, but for different job. I will merge this PR the `test` job is updated.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (autarch) [reply .github/workflows/ci.yml]: Done.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (autarch) [reply .github/workflows/ci.yml]: Done.

**Comment by autarch** [comment]
> Thanks for the update!
> 
> I had another look, and left 2 comments. Same comments as before, but for different job. I will merge this PR the `test` job is updated.

I fixed those. But FWIW, I think it'd be much better to merge the two jobs. As it stands, you're spinning up a new set of VMs for each job, which is _way_ slower than just running `cargo check` followed by `cargo test` in the same VM.

**Comment by waltzofpearls** [comment]
@autarch yeah, that makes sense. I will make that change after merging this PR.

**Pull request merged**